azkaban-common/src/main/java/azkaban/user/UserUtils.java

```diff
@@ -22,4 +37,67 @@ public static boolean hasPermissionforAction(final UserManager userManager, fina
 
     return false;
   }
+
+  /**
+   * Creates a watch thread which listens to specified files' modification and reloads
+   * configurations
+   */
+  static void setupWatch(final Map<String, ParseConfigFile> configFileMap) {
+    Preconditions.checkArgument(configFileMap != null);
+
+    Runnable runnable = () -> {
+      WatchService watchService;
+      Path path;
+      for (Map.Entry<String, ParseConfigFile> entry : configFileMap.entrySet()) {
+        String fileName = entry.getKey();
+        ParseConfigFile parser = entry.getValue();
+        Preconditions.checkArgument(fileName != null && parser != null);
+
+        final File file = new File(fileName);
+        if (!file.exists()) {
+          throw new IllegalArgumentException("User provided file " + fileName + " does not exist.");
```

Comment:
It would be good to still process the other files, even if one if missing or misconfigured.